.gitignore

```diff
@@ -0,0 +1,2 @@
+*.o
```

Comment:
you can use another build folder with CMake, it doesn't have to be in the source folder.

Reply:
I think this change is for Makefile only since NuttX Makefile based build system doesn't support the out of tree